### DISCUSSION TITLE
Run step-ca in kubernetes using PVC

### DISCUSSION THREAD

**Discussion by PavanKumar-MSK** [smallstep/certificates#1805] [Q&A]
I am trying to create a private CA using `step ca init` in Kubernetes using persistent volume for storing `certs`, `config`, `db`, and `secrets` so that my CA would be retained even if the pod crashes or restarts. 

All the files inside `certs`, `secrets` are storing in the PVC and are working as expected. The only problem is with the `db` folder. It is showing the following error-

> Error opening database of Type badgerv2: error opening Badger database: While syncing directory: /home/step/db.: sync /home/step/db: invalid argument.

If I use `emptyDir` instead of PVC for db, it is working fine but I really need to store `db` also in PVC.

What might be the issue here? It was working fine with hostPath PV in local using minikube. 

I don't know the configuration of the PV used. If the problem might be due to the PV configuration, any hint about a compatible configuration would also be helpful.

**Comment by hslatman** [comment]
Hey @PavanKumar-MSK, I don't know the specifics of the persistent volume or its configuration, but the error seems related to the file/directory not existing while a sync operations is being performed by Badger. Does the `/home/step/db` directory exist when you run the CA?

**Reply by PavanKumar-MSK** [reply]
Yes, it is being created when I run `step ca init`. The problem is when it needs to be accessed from my PV for writing the provisioner details into the database (I am using `--remote-management` flag). It is also not shown in my PV.
